tests/integration/test_merge_tree_blob_storage/configs/config.d/storage_conf.xml

```diff
@@ -0,0 +1,24 @@
+<clickhouse>
+    <storage_configuration>
+        <disks>
+            <blob_storage_disk>
+                <type>blob_storage</type>
+                <storage_account_url>http://azurite1:10000/</storage_account_url>
+                <container_already_exists>false</container_already_exists>
+                <skip_access_check>true</skip_access_check>
+                <!-- default credentials for Azurite storage account -->
+                <account_name>devstoreaccount1</account_name>
+                <account_key>Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==</account_key>
```

Comment:
With this configuration, when I run  integration tests, I get 
```
2021.12.01 22:00:59.151040 [ 59 ] {} <Error> void DB::SystemLog<DB::TraceLogElement>::flushImpl(const std::vector<LogElement> &, uint64_t) [LogElement = DB::TraceLogElement]: std::exception. Code: 1001, type: Azure::Storage::StorageException, e.what() = 400 Invalid storage account.
Invalid storage account.
RequestId:f36ca285-7d36-4c5a-a8a5-c9075399882f
Time:2021-12-01T19:00:57.849Z
0. /home/kssenii/workspace/CH-3/build-debug/../contrib/libcxx/include/exception:133: std::exception::capture() @ 0x15270e22 in /usr/bin/clickhouse
1. /home/kssenii/workspace/CH-3/build-debug/../contrib/libcxx/include/exception:111: std::exception::exception() @ 0x15270df3 in /usr/bin/clickhouse
2. /home/kssenii/workspace/CH-3/build-debug/../contrib/libcxx/src/support/runtime/stdexcept_default.ipp:33: std::runtime_error::runtime_error(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) @ 0x2a44ec9d in /usr/bin/clickhouse
3. /home/kssenii/workspace/CH-3/build-debug/../contrib/azure/sdk/core/azure-core/inc/azure/core/exception.hpp:78: Azure::Core::RequestFailedException::RequestFailedException(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) @ 0x26fe74c1 in /usr/bin/clickhouse
4. /home/kssenii/workspace/CH-3/build-debug/../contrib/azure/sdk/storage/azure-storage-common/inc/azure/storage/common/storage_exception.hpp:26: Azure::Storage::StorageException::StorageException(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) @ 0x270b9461 in /usr/bin/clickhouse
5. /home/kssenii/workspace/CH-3/build-debug/../contrib/azure/sdk/storage/azure-storage-common/src/storage_exception.cpp:144: Azure::Storage::StorageException::CreateFromResponse(std::__1::unique_ptr<Azure::Core::Http::RawResponse, std::__1::default_delete<Azure::Core::Http::RawResponse> >) @ 0x270b7609 in /usr/bin/clickhouse
6. /home/kssenii/workspace/CH-3/build-debug/../contrib/azure/sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/protocol/blob_rest_client.hpp:4760: Azure::Storage::Blobs::_detail::BlobRestClient::BlobContainer::Create(Azure::Core::Http::_internal::HttpPipeline&, Azure::Core::Url const&, Azure::Storage::Blobs::_detail::BlobRestClient::BlobContainer::CreateBlobContainerOptions const&, Azure::Core::Context const&) @ 0x27042ce7 in /usr/bin/clickhouse
7. /home/kssenii/workspace/CH-3/build-debug/../contrib/azure/sdk/storage/azure-storage-blobs/src/blob_container_client.cpp:147: Azure::Storage::Blobs::BlobContainerClient::Create(Azure::Storage::Blobs::CreateBlobContainerOptions const&, Azure::Core::Context const&) const @ 0x270409c8 in /usr/bin/clickhouse
8. /home/kssenii/workspace/CH-3/build-debug/../contrib/azure/sdk/storage/azure-storage-blobs/src/blob_service_client.cpp:235: Azure::Storage::Blobs::BlobServiceClient::CreateBlobContainer(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, Azure::Storage::Blobs::CreateBlobContainerOptions const&, Azure::Core::Context const&) const @ 0x27064886 in /usr/bin/clickhouse
9. /home/kssenii/workspace/CH-3/build-debug/../src/Disks/BlobStorage/registerDiskBlobStorage.cpp:177: DB::getBlobContainerClient(DB::BlobStorageEndpoint const&, Poco::Util::AbstractConfiguration const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) @ 0x215186fe in /usr/bin/clickhouse

```
I don't know azure blob storage and how to make a valid account.
But looks like it means it gets connected ok.

Reply:
I am about to check these connection issues with `Invalid storage account`, I am not sure yet that this is indeed a sign of connecting with Azurite. I am surprised it doesn't fail at `ListBlobContainers`, src/Disks/BlobStorage/registerDiskBlobStorage.cpp:171. I am also about to check the logs in the `_instances` directory.

I didn't notice this missing exception before, I think it is the reason why I thought I managed to connect to Azurite in the test.

I updated the Dockerfile to install `azure-storage-blob`.

I think I don't have the access rights to put the library in `ClickHouse-extras`, is that correct?

I have a new problem with running the tests with `./runner`, I get `FileNotFoundError: [Errno 2] No such file or directory: '/compose/docker_compose_azurite.yml'` that I will investigate now as well.

Reply:
> rights to put the library in ClickHouse-extras, is that correct?

Yes, I can fork azure blob storage libtary there and you can make a pull request there if you want. Not necessary though. 

> I have a new problem with running the tests with ./runner

btw, just in case you missed this don't forget to do this https://github.com/ClickHouse/ClickHouse/blob/master/tests/integration/README.md?plain=1#L73

> I have a new problem with running the tests with ./runner, I get FileNotFoundError: [Errno 2] No such file or directory: '/compose/docker_compose_azurite.yml' that I will investigate now as well.

Yes, you need to go to `docker/test/integration/runner/` and execute command you will see in the comment in the first line of Dockerfile there.

Reply:
I edited the comment above in the meantime, posting the update here:

Rebuilding the image locally fails - when I do it, there are issues with apt-update, it looks like there is no connection to the network within the container, I get e.g. `Temporary failure resolving 'ru.archive.ubuntu.com`. I wasn't able to modify the image and run it with ./runner, there were some errors. I needed to create the file from scratch using `os.system` in the test.py code.

Reply:
But I got the `Invalid storage account` error as well, I can investigate further now.

Reply:
> Rebuilding the image locally fails - when I do it, there are issues with apt-update, it looks like there is no connection to the network within the container, I get e.g. Temporary failure resolving 'ru.archive.ubuntu.com

oh, I recently started to get the same error. It gets fixed by `sudo service docker restart`